sklearn/neighbors/lof.py

```diff
@@ -267,7 +277,39 @@ def _decision_function(self, X):
                             X_lrd[:, np.newaxis])
 
         # as bigger is better:
-        return -np.mean(lrd_ratios_array, axis=1)
+        scores = -np.mean(lrd_ratios_array, axis=1)
+
+        if hasattr(self, 'offset_'):  # means that we're in testing
+            return scores + self.offset_  # to make value 0 special
+        else:  # we're in training
+            return scores
+
+    def _score_samples(self, X):
```

Comment:
We want a `score_samples` method for every anomaly detection algo (returning the "raw" decision function as defined in original papers). 
Here it has to be private for the same reason as decision function and predict are private.